File: Web/coffee/lib/service/coffee_service.dart
import 'package:coffee/model/coffee.dart';
import 'package:coffee/model/result.dart';
import 'package:coffee/util/dio_app.dart';
import 'package:dio/dio.dart';

Future<Result<List<Coffee>>> getCoffeeList() async {
  Response response = await appDio.get("/coffee/coffeeList");
  Result<List<Coffee>> result = Result.fromJson(response.data,
      (coffees) => (coffees as List).map((e) => Coffee.fromJson(e)).toList());
  return result;
}

Future<Result<String>> deleteCoffee(int coffeeId) async {
  Response response = await appDio
      .post("/coffee/deleteCoffee", queryParameters: {"coffeeId": coffeeId});
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<String>> addOrUpdateCoffee(Coffee coffee) async {
  Response response =
      await appDio.post("/coffee/addOrUpdateCoffee", data: coffee);
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<String>> updateStatus(int coffeeId, int status) async {
  Response response = await appDio.post("/coffee/updateStatus",
      queryParameters: {"coffeeId": coffeeId, "status": status});
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

File: Server/src/main/java/com/drink/coffee/service/CoffeeService.java
package com.drink.coffee.service;

import com.drink.coffee.model.Result;
import com.drink.coffee.pojo.Coffee;

import java.util.List;

public interface CoffeeService {
    /**
     * 获取咖啡列表
     *
     * @return 咖啡列表
     */
    Result<List<Coffee>> getCoffeeList();

    /**
     * 删除咖啡
     *
     * @param coffeeId 咖啡Id
     * @return 删除结果
     */
    Result<String> deleteCoffee(Integer coffeeId);

    /**
     * 添加或更新咖啡信息
     *
     * @param coffee 咖啡实体
     * @return 更新结果
     */
    Result<String> addOrUpdateCoffee(Coffee coffee);

    /**
     * 查看咖啡详情
     *
     * @param coffeeId 咖啡Id
     * @return 咖啡
     */
    Result<Coffee> selectCoffeeInfo(Integer coffeeId);

    /**
     * 更新上架还是下架
     *
     * @param coffeeId 咖啡Id
     * @param status   上架0、下架1
     * @return 更新结果
     */
    Result<String> updateStatus(Integer coffeeId, Integer status);
}
